utils/timed_version.go

```diff
@@ -73,6 +77,10 @@ func (g *timedVersionGenerator) Next() TimedVersion {
 	}
 }
 
+func (g *timedVersionGenerator) SetClockOffset(clockOffset time.Duration) {
+	g.clockOffset = clockOffset
```

Comment:
this and the access in `Next()` need to guarded by `g.mu` or we will get flagged by race detector

Reply:
oops.. I missed that.. good catch Paul!